### PR TITLE
switch to default model when receiving ResourceNotFoundException

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
https://github.com/aws/aws-toolkit-jetbrains/assets/96078566/06c63ea9-cfe4-435f-9e0d-6857d48785e1

```
2023-10-24 14:40:03,124 [3605204]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-10-24 14:40:09,452 [3611532]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.util.DefaultCodeWhispererFileContextProvider - Crossfile is not supporting kotlin
2023-10-24 14:40:09,456 [3611536]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererService - Calling CodeWhisperer service, trigger type: OnDemand
2023-10-24 14:40:09,458 [3611538]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus - Starting CodeWhisperer invocation
2023-10-24 14:40:10,034 [3612114]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererService - The provided customization arn:aws:codewhisperer:us-east-1:419795008606:customization/GW99HG749UPH is not found, will fallback to the default and retry generate completion
2023-10-24 14:40:10,042 [3612122]   INFO - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererService - SessionId: 98dbd276-277c-48cc-932f-1c9896a3dbcf, RequestId: 98dbd276-277c-48cc-932f-1c9896a3dbcf, Jetbrains IDE: IntelliJ IDEA 2023.2.2, IDE version: IU-232.9921.47, Filename: SsoSupport.kt, Left context of current line: fun getCredential, Cursor line: 26, Caret offset: 1073, Exception Type: AccessDeniedException, Recommendations: 
None
2023-10-24 14:40:10,045 [3612125]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus - Ending CodeWhisperer invocation
2023-10-24 14:40:10,045 [3612125]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.customization.CodeWhispererModelConfigurator - Invalidate customization arn: arn:aws:codewhisperer:us-east-1:419795008606:customization/GW99HG749UPH
2023-10-24 14:40:10,047 [3612127]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.util.DefaultCodeWhispererFileContextProvider - Crossfile is not supporting kotlin
2023-10-24 14:40:10,047 [3612127]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererService - Calling CodeWhisperer service, trigger type: OnDemand
2023-10-24 14:40:10,047 [3612127]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus - Starting CodeWhisperer invocation
2023-10-24 14:40:10,828 [3612908]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus - Set current CodeWhisperer invocation sessionId: 126f6166-7a5c-4ae1-ac80-5c12b264276c
2023-10-24 14:40:10,830 [3612910]   INFO - software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererService - SessionId: 126f6166-7a5c-4ae1-ac80-5c12b264276c, RequestId: 91e70bf0-56c3-445c-890f-4950a47ee599, Jetbrains IDE: IntelliJ IDEA 2023.2.2, IDE version: IU-232.9921.47, Filename: SsoSupport.kt, Left context of current line: fun getCredential, Cursor line: 26, Caret offset: 1073, Latency: 779.0, Recommendations: 
(credentialsProvider: CredentialProvider): Credential? {
    return credentialsProvider.getCredential()
}

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 1]
```diff
@@ -259,7 +260,8 @@ class CodeWhispererService {
 
                 if (
                     CodeWhispererConstants.Customization.customizationNotFoundExceptionPredicate(e) ||
```
Comment (zixlin7) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:262]: what does customizationNotFoundExceptionPredicate check for?

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:262]:
```
e is AccessDeniedException && (e.message?.contains(invalidCustomizationMessage, ignoreCase = true) ?: false)
```

ADE & e.messagecontains("not found")

Comment (zixlin7) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:262]: do we need this or we can just update this one to check for RNF?

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:262]: yea let me remove it

**Review by zixlin7 — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Pull request merged**